src/PrestaShopBundle/DataCollector/HookRegistry.php

```diff
@@ -93,8 +94,8 @@ public function hookedByModule(ModuleInterface $module)
     /**
      * A callback have been executed by the module during the Hook dispatch.
      *
-     * @param ModuleInterface $module
-     * @param $args array All arguments passed to the Module callback
+     * @param ModuleCore $module
```

Comment:
Is it mandatory to force-type the implementation rather than the interface ? This reduces the customization/extension capabilities of module and shop developers

Reply:
@matks Yes, it's mandatory because the property name is defined in `ModuleCore` and not in the interface.

Reply:
@matks Should I modify the interface or the parameter of the method ?

Reply:
In this PR scope, you are right, you are fixing some ambiguous code. Maybe in a future PR we could fix it in a better way by using an updated interface. However I don't want to burden the scope of this PR more.